ui/packages/atlasmap-core/src/services/mapping-management.service.ts

```diff
@@ -156,17 +156,33 @@ export class MappingManagementService {
   }
 
   /**
-   * Remove all mappings from the current session.
+   * Remove all mappings from the current session (UI and backend service).
    */
-  async removeAllMappings(): Promise<boolean> {
-    return new Promise<boolean>(async (resolve) => {
-      // TODO: check these non null operator on the mappings
-      for (const mapping of this.cfg.mappings!.getAllMappings(true)) {
-        this.cfg.mappings!.removeMapping(mapping);
-        this.deselectMapping();
-      }
-      await this.notifyMappingUpdated();
-      resolve(true);
+  removeAllMappings(): Promise<boolean> {
+    return new Promise<boolean>((resolve) => {
+      const url: string =
+        this.cfg.initCfg.baseAtlasServiceUrl +
+        'project/' +
+        this.cfg.mappingDefinitionId +
+        '/mapping';
+      this.api
+        .delete(url)
+        .then(async (res: Response) => {
+          this.cfg.logger!.debug(`Delete Mappings Response: ${res.ok}`);
+          for (const mapping of this.cfg.mappings!.getAllMappings(true)) {
+            this.cfg.mappings!.removeMapping(mapping);
```

Comment:
Could this be instead fetch the mapping definition from backend and refresh? isn't there already a routine doing it?

Reply:
I'd still need to clear the mappings from the mappings model - otherwise it would appear like the mappings are being reintroduced.  I could fetch the mappings after clearing the UI side mappings model just to double check.

Reply:
yeah only fetching mapping definition isn't enough, then it needs to refresh from what is received - deserialize the MappingDefinition and reflect to the ConfigModel.mappings. I thought we're already doing at somewhere, at least we do on initialiization.

Reply:
Yes - `notifyMappingUpdated()` will cause a full refresh.

Reply:
The reason is to reduce the logic from UI side and always rely on the backend for data consistency. Eventually every single action on UI should be notified to backend side and UI only asks updated mappings/documents to backend side and display them. We have to reduce those UI side state manipulating one-by-one, that is the primary motivation of the series of refactorings I've been working on recently.

Reply:
And once we have all the data&logic migrated to the Java side, we can reuse them in a offline tool as well - i.e. without UI, programmatically build an ADM file in Java.

Reply:
Okay - so specifically with this change - its calling the new backend endpoint to clear out just the mappings.  It's then clearing out the UI-side mapping info and refreshing.  Do I add a fetchMappings backend call to verify that the mappings are gone - then do the refresh?

Reply:
1. invoke backend to remove mappings
2. fetch updated (i.e. empty) mappings from backend
3. reflect the empty mappings
I thought the deserialize method in UI do #3 automatically? Then #2 and #3 could be reused for any operation thereafter. Does notifyMappingUpdated() already do that?

Reply:
If the UI side modification is needed, it means that the UI side of change overrides the backend change. Then backend remove mapping invocation is just a waste. Then later time it could be a source of a bug, hard to nail down which logic wins.

Reply:
`notifyMappingUpdated()` serializes, validates and pushes the mappings down to the backend, triggering a UI refresh.  It's necessary to clear out the UI mappings model.  I could extend the change I have to fetch the mappings from the backend (which we know to be empty).  The existing `notifyMappingUpdated()` will reflect the empty mappings back to the UI.  So from your list:
1.  done
2. not done currently but could add but this function's job is to delete all mappings so would be redundant with clearing mapping model
3. done via  `notifyMappingUpdated()`

Reply:
Ultimately the UI needs to tell the backend what the mappings are as defined by the mapping model (created by the user setting mappings).

Reply:
ah so `notifyMappingUpdated()` is actually to reflect from UI to backend. Then we should eventually replace it with the opposite, {No.2 + No.3} . The {No.2 + No.3} routine will be eventually shared with all the operations, like add mapping entry, add/delete field to/from mapping, etc. You might think it's redundant specifically for delete mappings, but doing same for every operation in the future will reduce the chance of bug significantly.

Reply:
Okay - so changing `notifyMappingUpdated()` from a push to a pull {No.2 + No.3} .  So - for this PR I'd would say it is correct and working with the given semantics and I can create a new issue that will tackle changing `notifyMappingUpdated()` semantics across the UI.  wdyt?

Reply:
I mean I'm trying to get this refactoring gradually. This time we add {No.2 + No.3} and only delete mappings will use it. Since other operations stay using notifyMappingUpdated(), it's temporary redundant anyway. But eventually we remove notifyMappingUpdated() and rely on {No.2 + No.3} at all time, then redundancy will be also removed.

Reply:
I'm OK with deferring the refactoring to the next issue, but can you work on it before another UI logic is added? :-D

Reply:
sure - it probably needs some more thought than a quick change anyway.

Reply:
OK then let this one goes in as-is.
Note that changing `notifyMappingUpdated()` from push to pull at once will be too big to do it at once, rather it'd be better to
1. Add pull routine
2. Let "remove mappings" use the pull routine and verify UI state modification is not necessary
3. Continue for other operations... many of existing mapping manipulation operations will need to migrate the logic from UI to the backend
4. Finally remove `notifyMappingUpdated()`

Reply:
ok - will create a new issue.